runServerSetup.sh

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env bash
+#
+# Checks if user has installed Docker Toolbox or Desktop and
+# Starts Application as appropriate
+
+detectOStype(){
+    if [[ `Uname` =~ "MING"? ]];then
+      USER_OS="Windows"
+    elif [[ `Uname` =~ "Darwin"? ]];then
+      USER_OS="macOS"
+    elif [[ `Uname` =~ "Linux"? ]];then
+      USER_OS="Linux"
+    else
+      USER_OS="unknown"
+    fi
+    Printf "User OS is: %s \n" ${USER_OS}
+}
+
+dockerIsToolboxOrDesktop() {
+  if [ "${DOCKER_TOOLBOX_INSTALL_PATH}" ];then
+    echo "Using Docker Toolbox YML for Docker-Compose"
+    DOCKER_COMPOSE_FILE=${DOCKER_TOOLBOX_YML}
+  else
+    dockerIsDesktop
+  fi
+}
+
+dockerIsDesktop() 
+{
+  for d in "${PATH}"; do
+    if [[ $d =~ "DockerDesktop"? ]]; then
+        echo "Using Docker Desktop. Default YML chosen for Docker-Compose"
+        DOCKER_COMPOSE_FILE=${DOCKER_DEFAULT_YML}
+    else
+        echo "Unable to detect Docker Installation, tries with Default YML"
+    fi
+  done
+}
+
+openServerPage()
+{
+    if [ "${USER_OS}" == "Windows" ];then
+      start chrome http://localhost:8000
```

Comment:
With Docker Toolbox, localhost does not seem to work. You will need to find the IP of the docker-machine. It's usually 192.168.99.100, can be determined `docker-machine ip Default`.

Reply:
Maybe `start chrome $(docker-machine ip default):$WEBPORT`, with WEBPORT=8000 defined somewhere for making changes easier later 😸